src/sentry/static/sentry/app/views/eventsV2/resultsChart.tsx

```diff
@@ -2,48 +2,127 @@ import React from 'react';
 import styled from '@emotion/styled';
 import * as ReactRouter from 'react-router';
 import {Location} from 'history';
+import isEqual from 'lodash/isEqual';
 
 import {Organization} from 'app/types';
-
+import {getUtcToLocalDateObject} from 'app/utils/dates';
+import {Client} from 'app/api';
+import {getParams} from 'app/components/organizations/globalSelectionHeader/getParams';
 import {Panel} from 'app/components/panels';
 import getDynamicText from 'app/utils/getDynamicText';
-import EventsChart from 'app/views/events/eventsChart';
+import {EventsChart} from 'app/views/events/eventsChart';
 
 import ChartFooter from './chartFooter';
 import EventView from './eventView';
 
-type Props = {
+type ResultsChartProps = {
+  api: Client;
   router: ReactRouter.InjectedRouter;
   organization: Organization;
   eventView: EventView;
   location: Location;
-
-  total: number | null;
-  onAxisChange: (value: string) => void;
 };
 
-export default class ResultsChart extends React.Component<Props> {
+class ResultsChart extends React.Component<ResultsChartProps> {
+  shouldComponentUpdate(nextProps: ResultsChartProps) {
+    const {eventView, ...restProps} = this.props;
+    const {eventView: nextEventView, ...restNextProps} = nextProps;
+
+    if (!eventView.isEqualTo(nextEventView)) {
+      return true;
+    }
+
+    return !isEqual(restProps, restNextProps);
+  }
+
   render() {
-    const {eventView, location, organization, router, total, onAxisChange} = this.props;
+    const {api, eventView, location, organization, router} = this.props;
 
     const yAxisValue = eventView.getYAxis();
 
+    const globalSelection = eventView.getGlobalSelection();
+    const start = globalSelection.start
+      ? getUtcToLocalDateObject(globalSelection.start)
+      : undefined;
+
+    const end = globalSelection.end
+      ? getUtcToLocalDateObject(globalSelection.end)
+      : undefined;
+
+    const {utc} = getParams(location.query);
+
     return (
-      <StyledPanel>
+      <React.Fragment>
         {getDynamicText({
           value: (
             <EventsChart
+              api={api}
               router={router}
               query={eventView.getEventsAPIPayload(location).query}
               organization={organization}
               showLegend
               yAxis={yAxisValue}
-              project={eventView.project as number[]}
-              environment={eventView.environment as string[]}
+              project={globalSelection.project as number[]}
+              environment={globalSelection.environment as string[]}
```

Comment:
Should getGlobalSelection() define types so you don't have to cast?

Reply:
fixed in https://github.com/getsentry/sentry/pull/17186/commits/fdeab6edae8c7e6059509167e081bd347804264f